ld-pruning-wf.wdl

```diff
@@ -0,0 +1,437 @@
+			py_filename = basename + "subset_chr" + chromosome + ".gds"
+
+		f.write("subset_gds_file " + py_filename)
+		f.close()
+		CODE
+
+		R -q --vanilla < /usr/local/analysis_pipeline/R/subset_gds.R --args subset_gds.config
+	}
+
+	runtime {
+		cpu: cpu
+		docker: "uwgac/topmed-master:2.10.0"
+		disks: "local-disk " + final_disk_dize + " HDD"
+		memory: "${memory} GB"
+		preemptibles: "${preempt}"
+	}
+	output {
+		File config_file = "subset_gds.config"
+		File subset_output = glob("*.gds")[0]
+	}
+}
+
+# [3] merge_gds -- merge a bunch of GDS files into one
+task merge_gds {
+	input {
+		Array[File] gdss
+
+		# having an input that depends upon a previous task's output reigns in
+		# cromwell's tendencies to run tasks out of order
+		File? enforce_chronological_order
+
+		# optional
+		String? out_prefix
+
+		# runtime attributes
+		Int addldisk = 5
+		Int cpu = 2
+		Int memory = 4
+		Int preempt = 3
+	}
+
+	# Estimate disk size required
+	Int gds_size = ceil(size(gdss, "GB"))
+	Int final_disk_dize = gds_size * 3 + addldisk
+	String filename = select_first([out_prefix, "merged"])
+
+	command <<<
+		set -eux -o pipefail
+
+		# CWL has an ln -s, will probably need to use copy trick again
+		echo "Copying inputs into the workdir"
+		BASH_FILES=(~{sep=" " gdss})
+		for BASH_FILE in ${BASH_FILES[@]};
+		do
+			cp ${BASH_FILE} .
+		done
+
+		echo "Generating config file"
+		python << CODE
+		import os
+
+		def find_chromosome(file):
+			chr_array = []
+			chrom_num = split_on_chromosome(file)
+			if(unicode(str(chrom_num[1])).isnumeric()):
+				# two digit number
+				chr_array.append(chrom_num[0])
+				chr_array.append(chrom_num[1])
+			else:
+				# one digit number or Y/X/M
+				chr_array.append(chrom_num[0])
+			return "".join(chr_array)
+
+		def split_on_chromosome(file):
+			chrom_num = file.split("chr")[1]
+			return chrom_num
+
+		def write_config(chrs, path):
+			f = open("merge_gds.config", "a")
+
+			if "~{out_prefix}" != "":
+				merged_gds_file_name = "~{out_prefix}" + ".gds"
+			else:
+				merged_gds_file_name = "merged.gds"
+
+			f.write('gds_file "' + path[0] + 'chr ' + path[1] + '"\n')
+			f.write('merged_gds_file "' + merged_gds_file_name + '"\n')
+			f.write('chromosomes "' + chrs + '"\n')
+
+			f.close()
+
+		gds_array_fullpath = ['~{sep="','" gdss}']
+		gds_array_basenames = []
+		for fullpath in gds_array_fullpath:
+			gds_array_basenames.append(os.path.basename(fullpath))
+
+		a_file = gds_array_basenames[0]
+		chr = find_chromosome(a_file)
+		path = a_file.split('chr'+chr)
+
+		# make list of all chromosomes found in input files
+		chr_array = []
+		for gds_file in gds_array_basenames:
+			chrom_num = find_chromosome(gds_file)
+			chr_array.append(chrom_num)
+		chrs = ' '.join(chr_array)
+
+		write_config(chrs, path)
+
+		exit()
+		CODE
+
+		Rscript /usr/local/analysis_pipeline/R/merge_gds.R merge_gds.config
+	>>>
+
+	runtime {
+		cpu: cpu
+		disks: "local-disk " + final_disk_dize + " HDD"
+		docker: "uwgac/topmed-master:2.10.0"
```

Comment:
At a future date we probably want to set this as a parameter. That way you can give it a default value, but then you can pass different values w/ the checker workflow!

Reply:
Do you mean setting final_disk_size as a parameter? Right now you can already use the addldisk parameter to throw extra gigabytes into your calculation.

Reply:
No, I meant the Docker container.

Reply:
You are right, putting the container as a parameter is bad practice:
https://docs.dockstore.org/en/develop/advanced-topics/best-practices/best-practices-secure-fair-workflows.html

It looks like the prize for winning the pie eating contest is... more pie. Time to change the Docker container string to include a sha256.

Reply:
More pie has been added. 🥧